network-store-model/src/main/java/com/powsybl/network/store/model/Resource.java

```diff
@@ -65,75 +80,88 @@ public Builder(ResourceType type) {
             if (attributes == null) {
                 throw new IllegalStateException("attributes is not set");
             }
-            return new Resource<>(type, id, attributes);
+
```

Comment:
having dual behaviors like this makes the system more complex and less robust, if we can eliminate it we should..